### PR TITLE
Upgrade Ruma

### PR DESCRIPTION
**Pull request by jplatte**
I think this will fail CI because it has some weird failures locally, but let's see.

### CODE REVIEW

**Comment by poljar** [comment]
There's already a bump Ruma commit that fixes the issues over here: https://github.com/matrix-org/matrix-rust-sdk/pull/1362/commits/7da8f311374feaa71549823a6c5785399cd8e5c2.

**Comment by jplatte** [comment]
Oohhhh thanks! I'll base my PR on yours then.

**Comment by poljar** [comment]
> Oohhhh thanks! I'll base my PR on yours then.

Feel free to cherry-pick the commit, not sure when that PR will get merged.

**Comment by jplatte** [comment]
Now depends on https://github.com/ruma/ruma/pull/1457.

File: crates/matrix-sdk-base/src/rooms/mod.rs [hunk 1]
```diff
@@ -158,7 +159,15 @@ impl BaseRoomInfo {
...
             AnyStrippedStateEvent::RoomEncryption(encryption) => {
-                self.encryption = Some(encryption.content.clone());
+                if let Some(algorithm) = &encryption.content.algorithm {
+                    let content = assign!(RoomEncryptionEventContent::new(algorithm.clone()), {
+                        rotation_period_ms: encryption.content.rotation_period_ms,
+                        rotation_period_msgs: encryption.content.rotation_period_msgs,
+                    });
+                    self.encryption = Some(content);
+                } else {
+                    // FIXME: encryption event exists but is redacted
```
Comment (poljar) [inline crates/matrix-sdk-base/src/rooms/mod.rs:169]: In that case, let's set the default rotation periods.

Comment (jplatte) [reply crates/matrix-sdk-base/src/rooms/mod.rs:169]: That's not the problem, the problem is unknown algorithm.

**Review by poljar — APPROVED** [review]

**Pull request merged**